core/connection/tlsconfig_clone_17.go

```diff
@@ -6,7 +6,7 @@ import "crypto/tls"
 // used concurrently by a TLS client or server.
 func (c *TLSConfig) Clone() *TLSConfig {
 	cfg := cloneconfig(c.Config)
-	return &TLSConfig{cfg}
+	return &TLSConfig{cfg, nil}
```

Comment:
I think we'll need to clone the `clientCertPass` here as well since we always clone the TLSConfig before use.